### PR TITLE
feat: support download base mirror

### PR DESCRIPTION
**Pull request by fengmk2**
Use `SAUCECTL_INSTALL_BINARY_BASE_MIRROR` env to set download base site

e.g.: https://github.com/cnpm/mirrors/pull/275

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +27,14 @@ saucectl version 0.4.0
 
 __Note:__ if you run the command for the first time it will initially download the binary for you. This only happens once.
 
+### Install binary from Mirror
+
+Use the `SAUCECTL_INSTALL_BINARY_BASE_MIRROR` env override the default download base site (https://github.com/saucelabs/saucectl/releases/download)
```
Comment (alexplischke) [inline README.md]: I think we can omit the 'base' part from it while still remaining intuitive.

```suggestion
Use the `SAUCECTL_INSTALL_BINARY_MIRROR` env to override the default download base site (https://github.com/saucelabs/saucectl/releases/download)
```

File: README.md [hunk 2]
```diff
@@ -27,6 +27,14 @@ saucectl version 0.4.0
...
+
+```bash
+SAUCECTL_INSTALL_BINARY_BASE_MIRROR=https://your-mirror-download-site.com/foo/bar npm i -g saucectl
```
Comment (alexplischke) [inline README.md]:
```suggestion
SAUCECTL_INSTALL_BINARY_MIRROR=https://your-mirror-download-site.com/foo/bar npm i -g saucectl
```

File: __tests__/main.test.js [hunk 3]
```diff
@@ -35,5 +35,12 @@ describe('main', function () {
 			expect(bw._src).toMatchSnapshot();
 			expect(typeof bw._dest).toEqual('string');
 		});
+
+		test('should respect the SAUCECTL_INSTALL_BINARY_BASE_MIRROR wrapper', async function () {
```
Comment (alexplischke) [inline __tests__/main.test.js]:
```suggestion
		test('should respect the SAUCECTL_INSTALL_BINARY_MIRROR wrapper', async function () {
```

File: index.js [hunk 4]
```diff
@@ -42,7 +43,7 @@ async function main (b, args) {
 
 /* istanbul ignore if */
 if (require.main === module) {
-    const bw = binWrapper(process.env.SAUCECTL_INSTALL_BINARY);
+    const bw = binWrapper(process.env.SAUCECTL_INSTALL_BINARY, process.env.SAUCECTL_INSTALL_BINARY_BASE_MIRROR);
```
Comment (alexplischke) [inline index.js]:
```suggestion
    const bw = binWrapper(process.env.SAUCECTL_INSTALL_BINARY, process.env.SAUCECTL_INSTALL_BINARY_MIRROR);
```

File: install.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ const { binWrapper } = require('./index.js');
 // execution.
 async function install () {
   console.info('Fetching saucectl binary');
-  const bw = binWrapper(process.env.SAUCECTL_INSTALL_BINARY);
+  const bw = binWrapper(process.env.SAUCECTL_INSTALL_BINARY, process.env.SAUCECTL_INSTALL_BINARY_BASE_MIRROR);
```
Comment (alexplischke) [inline install.js]:
```suggestion
  const bw = binWrapper(process.env.SAUCECTL_INSTALL_BINARY, process.env.SAUCECTL_INSTALL_BINARY_MIRROR);
```

**Review by alexplischke — COMMENTED** [review]
Thank you! I made a small suggestion regarding the env var naming, LGTM otherwise!

**Comment by fengmk2** [comment]
@alexplischke Nice :)

**Pull request merged**